internal/p2p/p2ptest/require.go

```diff
@@ -95,11 +96,14 @@ func RequireSendReceive(
 }
 
 // RequireNoUpdates requires that a PeerUpdates subscription is empty.
-func RequireNoUpdates(t *testing.T, peerUpdates *p2p.PeerUpdates) {
+func RequireNoUpdates(ctx context.Context, t *testing.T, peerUpdates *p2p.PeerUpdates) {
 	t.Helper()
 	select {
 	case update := <-peerUpdates.Updates():
-		require.Fail(t, "unexpected peer updates", "got %v", update)
+		if ctx.Err() == nil {
+			require.Fail(t, "unexpected peer updates", "got %v", update)
```

Comment:
I think we'd need this to be `require.Failf` if we want formatting.

Reply:
```
func Fail(t TestingT, failureMessage string, msgAndArgs ...interface{}) bool {
	if h, ok := t.(tHelper); ok {
		h.Helper()
	}
	content := []labeledContent{
		{"Error Trace", strings.Join(CallerInfo(), "\n\t\t\t")},
		{"Error", failureMessage},
	}

	// Add test name if the Go version supports it
	if n, ok := t.(interface {
		Name() string
	}); ok {
		content = append(content, labeledContent{"Test", n.Name()})
	}

	message := messageFromMsgAndArgs(msgAndArgs...)
	if len(message) > 0 {
		content = append(content, labeledContent{"Messages", message})
	}

	t.Errorf("\n%s", ""+labeledOutput(content...))

	return false
}
```

looks like it works